src/main.rs

```diff
@@ -85,6 +85,36 @@ impl From<&BpfProgram> for Row<'_> {
     }
 }
 
+/// Responsible for managing the terminal state and cleaning up when the program exits
+struct TerminalManager {
+    terminal: Terminal<CrosstermBackend<Stdout>>,
+}
+
+impl TerminalManager {
+    fn new() -> Result<Self> {
+        enable_raw_mode()?;
+        let mut stdout = io::stdout();
+        execute!(stdout, EnterAlternateScreen)?;
+        let backend = CrosstermBackend::new(stdout);
+        let terminal = Terminal::new(backend)?;
+        Ok(Self { terminal })
+    }
+}
+
+impl Drop for TerminalManager {
+    fn drop(&mut self) {
+        disable_raw_mode().unwrap_or_else(|e| eprintln!("Error disabling raw mode: {:?}", e));
+        execute!(self.terminal.backend_mut(), LeaveAlternateScreen)
```

Comment:
Should swap this with the previous line, I think. Need to send the ANSI resource sequence before disabling raw mode - or at least before bailing if raw mode fails.

Reply:
Swapped